### PR TITLE
Feature/input with hint

### PR DESCRIPTION
**Pull request by straleyb**
Adds input classes and factory for those classes. Adds the hint into the generation of the simple form.

### CODE REVIEW

File: app/forms/generic_asset_form.rb [hunk 1]
```diff
@@ -2,8 +2,11 @@ class GenericAssetForm
   include HydraEditor::Form
   # TODO: Put :workType back in once our editor can support URLs and/or we have
   # a "clean" graph from AF
+  #
+
```
Comment (jechols) [inline app/forms/generic_asset_form.rb]: Kill these unnecessary blank lines

**Comment by jechols** [comment]
We will have to do one of the following:

- Wait for the HydraEditor merge
- Update the Gemfile to attach to my repo: https://github.com/jechols/hydra-editor/tree/feature/configurable-field-generator
- Fork HydraEditor here in OregonDigital's github namespace and update the Gemfile to attach to that

Until one of these things happens, even if tests pass we won't have the necessary infrastructure for this code to run

**Comment by jechols** [comment]
Decision reached: point to jechols repo until we get the hydra editor PR merged

File: app/forms/oregon_digital/fields/input.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          required: required?,
+          as: field_type,
+          input_html: input_html_options,
+          wrapper_html: wrapper_html_options,
+          hint: hint,
+        }
+      end
+
+      def options
+        all_options.select {|option, method| method != nil }
```
Comment (terrellt) [inline app/forms/oregon_digital/fields/input.rb]: all_options.compact

**Pull request merged**